pyfr/plugins/integrate.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+
+import re
+
+import numpy as np
+
+from pyfr.mpiutil import get_comm_rank_root, get_mpi
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin, init_csv
+from pyfr.quadrules import get_quadrule
+
+
+class IntegratePlugin(BasePlugin):
+    name = 'integrate'
+    systems = ['*']
+    formulations = ['dual', 'std']
+
+    def __init__(self, intg, cfgsect, suffix=None):
+        super().__init__(intg, cfgsect, suffix)
+
+        comm, rank, root = get_comm_rank_root()
+        
+        # Underlying elements class
+        self.elementscls = intg.system.elementscls
+
+        # Expressions to integrate
+        c = self.cfg.items_as('constants', float)
+        self.exprs = [self.cfg.getexpr(cfgsect, k, subs=c)
+                      for k in self.cfg.items(cfgsect)
+                      if k.startswith('int-')]
+
+        # Gradient pre-processing
+        self._init_gradients(intg)
+
+        # Save a reference to the physical solution point locations
+        self.plocs = intg.system.ele_ploc_upts
+
+        # Integration parameters
+        self.nsteps = self.cfg.getint(cfgsect, 'nsteps')
+        
+        # The root rank needs to open the output file
+        if rank == root:
+            header = ['t'] + [k for k in self.cfg.items(cfgsect) if \
```

Comment:
No need for \ in Python if we're inside brackets.  Normally if statement is brought down too for clarity.